### PR TITLE
replace deprecated policy configuration with peerAuthentication in zh

### PR DESCRIPTION
**Pull request by zirain**
`policy` and `meshpolicy` are removed since istio 1.6.0, see: [#22602](https://github.com/istio/istio/issues/22602)
we need to update the troubleshooting doc for the example configuration.

resolve: [istio/istio#24502](https://github.com/istio/istio/issues/24502)

[ ] Configuration Infrastructure
[X ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by zirain** [comment]
> googlebot I signed

@googlebot I signed it!

File: content/zh/docs/tasks/security/authentication/mtls-migration/index.md [hunk 1]
```diff
@@ -102,17 +102,13 @@ sleep.legacy to httpbin.foo: 200
...
-  name: "example-httpbin-strict"
-  namespace: foo
+  name: "default"
 spec:
-  targets:
-  - name: httpbin
-  peers:
-  - mtls:
-      mode: STRICT
+  ntls:
```
Comment (SataQiu) [inline content/zh/docs/tasks/security/authentication/mtls-migration/index.md:110]:
```suggestion
  mtls:
```

Comment (zirain) [reply content/zh/docs/tasks/security/authentication/mtls-migration/index.md]: thx @SataQiu 
it's fixed now!

**Comment by zirain** [comment]
> @zirain: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> k8s-tests_istio.io	[8257334](https://github.com/istio/istio.io/commit/8257334712b3d756368db52569f84d3d24755209)	[link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/7834/k8s-tests_istio.io/2839)	`/test k8s-tests_istio.io`

/retest

**Pull request merged**